City-of-Helsinki/drupal-helfi-platform-config
# Uhf 4465 linked events frontend

# Frontend functionality for LinkedEvents-paragraph [UHF-4465](https://helsinkisolutionoffice.atlassian.net/browse/UHF-4465)
Add frontend-capabilities to LinkedEvents-integration

## What was done
- Added JS to handle data retrieval from api and render it
- Added tranlations

## How to install
- Run `composer reinstall drupal/helfi_platform_config --prefer-source` in an instance of your choice
- Cd into module directory and checkout to branch
- `make shell` into container and run `drush en helfi_events -y; drush locale:check; drush locale:update`
- Check instructions for enabling styles [here](https://github.com/City-of-Helsinki/drupal-hdbt/pull/290).

## How to test
- Create or modify a Landing or basic page node
- Add an events paragraph to the page.
- Make a search for events here: https://tapahtumat.hel.fi/. Add the resulting url as value to 'Api Url' field
- View the page. The paragraph should construct a list of events identical to the one on https://tapahtumat.hel.fi/.
- Go back and edit the Api url value to be gibberish or modify a legit link to result in zero results. This should render a view informing the user that no events were found.
- Compare to this: https://app.abstract.com/projects/6a2238d2-fcb7-40e4-b2aa-a4487941ad8a/branches/1906cb8e-e1bb-4fa6-b5c8-22474c879276/collections/71b484aa-eddc-449e-a64b-5d495c402590.

## Other PRs
- https://github.com/City-of-Helsinki/drupal-hdbt/pull/290
